ovs/lib/scheduledtask.py

```diff
@@ -195,186 +199,209 @@ def make_timestamp(offset):
+
+                        # Check MDS master is local. Trigger MDS handover if necessary
+                        configs = _verify_mds_config(current_vdisk=vdisk)
+                        storagedriver = StorageDriverList.get_by_storagedriver_id(vdisk.storagedriver_id)
+                        if configs[0].get('ip') != storagedriver.storagerouter.ip:
+                            ScheduledTaskController._logger.debug('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - MDS master is not local, trigger handover'.format(vpool.name, storagerouter.name, vdisk.name))
+                            MDSServiceController.ensure_safety(vdisk)
+                            configs = _verify_mds_config(current_vdisk=vdisk)
+                            if configs[0].get('ip') != storagedriver.storagerouter.ip:
+                                ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - Skipping because master MDS still not local'.format(vpool.name, storagerouter.name, vdisk.name))
+                                continue
+
+                        # Do the actual scrubbing
+                        with vdisk.storagedriver_client.make_locked_client(str(vdisk.volume_id)) as locked_client:
+                            ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - Retrieve and apply scrub work'.format(vpool.name, storagerouter.name, vdisk.name))
+                            work_units = locked_client.get_scrubbing_workunits()
+                            for work_unit in work_units:
+                                res = locked_client.scrub(work_unit=work_unit,
+                                                          scratch_dir=scrub_directory,
+                                                          log_sinks=[sink],
+                                                          backend_config=Configuration.get_configuration_path(backend_config_key))
+                                locked_client.apply_scrubbing_result(scrubbing_work_result=res)
+                            if work_units:
+                                ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - {3} work units successfully applied'.format(vpool.name, storagerouter.name, vdisk.name, len(work_units)))
+                            else:
+                                ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - vDisk {2} - No scrubbing required'.format(vpool.name, storagerouter.name, vdisk.name))
+            except Empty:  # Raised when all items have been fetched from the queue
+                ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - Queue completely processed'.format(vpool.name, storagerouter.name))
+            except Exception:
```

Comment:
The generic ```Exception``` should be caught inside the ```while True:``` so it will continue with the next disk if one fails.